### PR TITLE
feat: market update ACs for mark price and perps price

### PR DESCRIPTION
**Pull request by davidsiska-vega**

### CODE REVIEW

File: protocol/0053-PERP-product_builtin_perpetual_future.md [hunk 1]
```diff
@@ -364,3 +364,5 @@ Assert that the scaling factor is applied before the funding cap is applied (<a
 Assert all funding payments are correct when a perpetual market is suspended and then terminated via governance. (<a name="0053-PERP-041" href="#0053-PERP-041">0053-PERP-041</a>).
 
 The upper and lower clamp values are being correctly validated as per the [parameters defined in the spec](https://github.com/vegaprotocol/specs/blob/palazzo/protocol/0053-PERP-product_builtin_perpetual_future.md#1-product-parameters). (<a name="0053-PERP-042" href="#0053-PERP-042">0053-PERP-042</a>).
+
+Launch a perpetual futures market which uses the "Last Traded Price" for the "vega side price" for funding calculation. Submit a market update proposal to change this to a composite price with a configuration which uses the impact notional price from the order book. Observe that the new methodology for funding calculations is applied correctly from enactment onwards. (<a name="0053-PERP-043" href="#0053-PERP-043">0053-PERP-043</a>).
```
Comment (wwestgarth) [inline protocol/0053-PERP-product_builtin_perpetual_future.md]: Given its possible for `internalCompositePrice` to be set to `nil`. Can we have two more AC's one that takes a perptual market where the composite-price-config it is set to `nil` in its original proposal, and updates it to have a a composite-price configuration that is non-nil, and another that takes a market that was originally proposed with a composite-price-configuration and updates it to be set to `nil`?

**Review by wwestgarth — DISMISSED** [review]

File: protocol/0009-MRKP-mark_price.md [hunk 2]
```diff
@@ -132,6 +132,11 @@ Note that for calculating the median with an even number of entries we sort, pic
 
   - When `markPriceConfiguration: source weight` is provided then it must not be all `0` (<a name="0009-MRKP-059" href="#0009-MRKP-059">0009-MRKP-059</a>)
 
+  - Launch a cash settled futures market which uses the "Last Traded Price" mark price method. Submit a market update proposal to change this to a composite price with some chosen configuration. Observe that the new methodology is applied correctly from enactment onwards. (<a name="0009-MRKP-060" href="#0009-MRKP-060">0009-MRKP-060</a>).
```
Comment (gordsport) [inline protocol/0009-MRKP-mark_price.md]:
```suggestion
  - Launch a cash settled futures market which uses the "Last Traded Price" mark price method. Submit a market update proposal to change this to a composite price with some chosen configuration. Observe that the new methodology is applied correctly from enactment onwards. (<a name="0009-MRKP-133" href="#0009-MRKP-133">0009-MRKP-133</a>).
```

File: protocol/0009-MRKP-mark_price.md [hunk 3]
```diff
@@ -132,6 +132,11 @@ Note that for calculating the median with an even number of entries we sort, pic
...
+
+- Launch a perpetual futures market which uses the "Last Traded Price" mark price method. Submit a market update proposal to change this to a composite price with some chosen configuration. Observe that the new methodology is applied correctly from enactment onwards. (<a name="0009-MRKP-061" href="#0009-MRKP-061">0009-MRKP-061</a>).
```
Comment (gordsport) [inline protocol/0009-MRKP-mark_price.md]:
```suggestion
- Launch a perpetual futures market which uses the "Last Traded Price" mark price method. Submit a market update proposal to change this to a composite price with some chosen configuration. Observe that the new methodology is applied correctly from enactment onwards. (<a name="0009-MRKP-134" href="#0009-MRKP-134">0009-MRKP-134</a>).
```

Comment (davidsiska-vega) [reply protocol/0009-MRKP-mark_price.md]: Thanks! I've done that locally.

**Review by gordsport — APPROVED** [review]

**Pull request merged**